src/js/_enqueues/wp/image-mime-fallback.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @output wp-includes/js/wp-image-mime-fallback.js
+ */
+
+window.wp = window.wp || {};
+
+( function( document, settings ) {
+	settings = settings || {};
+	window.wp.imageMimeFallback = function( media ) {
+		for ( var i = 0; i < media.length; i++ ) {
+			try {
+				var image         = media[ i ],
+					media_details = image.media_details,
+					sizes         = media_details.sizes,
+					sizes_keys    = Object.keys( sizes );
+
+				var images = document.querySelectorAll( 'img.wp-image-' + image.id );
+				for ( var j = 0; j < images.length; j++ ) {
+
+					var src = images[ j ].src;
+
+					// If there are no sizes, there is nothing more to replace.
+					if ( ! sizes_keys.length ) {
+						continue;
+					}
+
+					var srcset = images[ j ].getAttribute( 'srcset' );
+
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
+						var original_size_key  = sizes_keys[ k ],
+							original_sizes     = sizes[ original_size_key ],
+							original_url       = original_sizes.source_url;
+
+						if ( original_url.match(/\.(jpeg|jpg|jpe)$/) ) {
+							continue;
+						}
+
+						var get_update_image      = '',
+							original_file         = original_sizes.file,
+							original_filesize     = original_sizes.filesize,
+							original_aspect_ratio = original_sizes.width / original_sizes.height;
+
+						// Check to see if the image src has any size set, then update it.
+						if ( original_url === src ) {
+							get_update_image = replaceImageUrl( media_details, original_size_key, original_aspect_ratio, original_filesize );
+
+							src = src.replace( original_file, get_update_image );
+
+							// If there is no srcset and the src has been replaced, there is nothing more to replace.
+							if ( ! srcset ) {
+								break;
+							}
+						}
+
+						if ( srcset ) {
+							get_update_image = replaceImageUrl( media_details, original_size_key, original_aspect_ratio, original_filesize );
+
+							srcset = srcset.replace( original_file, get_update_image );
+						}
+					}
+
+					if ( srcset ) {
+						images[ j ].setAttribute( 'srcset', srcset );
+					}
+
+					if ( src ) {
+						images[ j ].setAttribute( 'src', src );
+					}
+				}
+			} catch ( e ) {
+			}
+		}
+	};
+
+	var replaceImageUrl = function( media_details, original_size_key, original_aspect_ratio, original_filesize ) {
+		var sizes      = media_details.sizes,
+			sizes_keys = Object.keys( sizes );
+
+		for ( var i = 0; i < sizes_keys.length; i++ ) {
+			var src      = media_details.original_image ? media_details.original_image : sizes.full.file,
+				size_key = sizes_keys[ i ],
+				size     = sizes[ size_key ];
+
+			if ( size_key !== original_size_key ) {
+				var filesize     = size.filesize,
+					aspect_ratio = size.width / size.height;
+
+				if ( aspect_ratio === original_aspect_ratio && filesize > original_filesize ) {
+					src = size.file;
+				}
+			}
+			return src;
```

Comment:
I think there are some things off in the logic here. For example, why are you setting `src` unconditionally to the original/full image _within_ the `for` loop through the `sizes`? Shouldn't this be set before the `for` loop? Similarly, shouldn't the return statement be at the end of the function rather than inside the `for` loop?

Overall, I think we can also make some performance optimizations here:
* Instead of going through all the `sizes` for every individual image, we could sort the `sizes` once for each `image`, by the size's `width` or `height` so that smaller ones come first.
* This way, when we look at a single image, since we know the `original_size_key`, we can start only going through the array starting with the size key after that (since we know everything before that will be smaller images).
* We could then even avoid some checks, since at that point basically the first image we find that has the same aspect ratio would be the optimal replacement.

Reply:
I have update PR with the changes. Can you please re-review it?